### PR TITLE
CI: Pylint to ruff

### PR DESCRIPTION
**Pull request by jnewb1**

### CODE REVIEW

File: tools/lib/tests/test_caching.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 import unittest
 
 os.environ["COMMA_CACHE"] = "/tmp/__test_cache__"
-from tools.lib.url_file import URLFile, CACHE_DIR
+from tools.lib.url_file import URLFile, CACHE_DIR # noqa: E402
```
Comment (adeebshihadeh) [inline tools/lib/tests/test_caching.py:7]: we can disable E402

File: selfdrive/car/nissan/values.py [hunk 2]
```diff
@@ -45,7 +46,6 @@ class NissanCarInfo(CarInfo):
 }
 
 FINGERPRINTS = {
-  # pylint: disable=C0301
```
Comment (adeebshihadeh) [inline selfdrive/car/nissan/values.py:48]: how is this handled now?

Comment (jnewb1) [reply selfdrive/car/nissan/values.py:48]: ruff doesn't allow disabling blocks of code like pylint. I made ruff exclude E501 by putting
```# ruff: noqa: E501``` to the top of the file

only other alternative I see is adding that line to the end of every fingerprint

File: tools/lib/tests/test_caching.py (hunk 1 above)
Comment (jnewb1) [reply tools/lib/tests/test_caching.py]: done

File: selfdrive/car/nissan/values.py (hunk 2 above)
Comment (jnewb1) [reply selfdrive/car/nissan/values.py:48]: don't see any regex comparing ability in ruff

**Pull request merged**